src/src/components/Contexts/SettingsContext.tsx

```diff
@@ -0,0 +1,51 @@
+import React from 'react';
+import { ajax } from '../../utils/ajax';
+import { AuthContext } from './AuthContext';
+
+type UserSettings = {
+  readonly weekStart: number;
+};
+
+export function SettingsProvider({
+  children,
+}: {
+  readonly children: React.ReactNode;
+}): JSX.Element {
+  const authContext = React.useContext(AuthContext);
+  const [weekStart, setWeekStart] = React.useState<number>(0);
+  const [found, setFound] = React.useState<boolean>(false);
+
+  // If the token has been set, try an update
+  // Also make sure we havent already fetched
+  if (authContext.token && !found) {
+    ajax(
+      'https://www.googleapis.com/calendar/v3/users/me/settings/weekStart'
+    ).then((res) => {
+      res.json().then((val) => {
+        if (typeof val['value'] === 'string') {
+          let wstart = new Number(val['value']) as number;
```

Comment:
use `Number.parseInt()` instead of `new Number`

Reply:
This is causing a bug where weekStart is actually interpreted as an empty object, rather than a number (because `new Number()` creates an object)

<img width="200" alt="Screenshot 2023-03-04 at 18 22 19" src="https://user-images.githubusercontent.com/40512816/222938154-03d19963-3b20-4b42-8f92-b26b36a5c2bb.png">

Reply:
After this change can confirm that the code is working like intended